### ISSUE TITLE
False negative reportMissingTypeArgument in class instantiation

### ISSUE THREAD

**Issue by otonnesen** [microsoft/pyright#3451]
**Describe the bug**
Missing type arguments aren't caught in class instantiations.

Note: this is arguably a feature request rather than a bug, since instantiating a generic without a type argument is technically valid (and a workaround to get more helpful linting is simply annotating the variable directly).

**To Reproduce**
```toml
[tool.pyright]
typeCheckingMode = "basic"
reportMissingTypeArgument = "information"
```

```python
from typing import Generic, TypeVar

A = TypeVar("A")


class Foo(Generic[A]):
    pass


foo: Foo = Foo()  # Expected type arguments for generic class "Foo" (expected) 
bar = Foo()  # OK (not expected)
baz: Foo[int] = Foo()  # OK (expected)
quux = Foo[int]()  # OK (expected)
```

**Expected behavior**
`bar = Foo()` should be flagged since it's missing a type argument.

**VS Code extension or command-line**
CLI version: `pyright 1.1.246`

**Comment by erictraut** [comment]
The `reportMissingTypeArgument` diagnostic rule is designed to flag missing type arguments in type expressions — in particular, type annotations that involve the use of a generic class or generic type alias. It is not meant to indicate that a type variable was unsolved in a constructor call (or in a call to any function more generally). The check is working as designed.

Pyright will report Unknown type arguments if you enable "strict" type checking mode or, more specifically, enable the diagnostic rules whose names begin with `reportUnkown...`. You can see this if you add a `# pyright: strict` comment at the start of your code sample above.

**Comment by otonnesen** [comment]
Makes sense, appreciate the quick response!
